### ISSUE TITLE
[Bug] BetterDiscord uninstalls with every Discord update

### ISSUE THREAD

**Issue by ItsCubeTime** [BetterDiscord/BetterDiscord#1691]
### Before Reporting

- [X] I found no existing issues matching my bug
- [X] My issue is not caused by a plugin or theme

### Describe the Bug

Close Discord > Open Discord > Let Discords automatic update wizard run > Go to client settings at bottom left (cogwheel) > Scroll down in the menu to the left > Notice that all BetterDiscord entries are no longer present (and that all other BetterDiscord related functionality is no longer  functional).

Currently (afaik), as a user you will now need to re-run BetterDiscords install wizard to regain BetterDiscord functionality, installed plugins and themes appear to be unaffected, but you need to manually reactivate them (which on its own is a lot of labor to go through).

### Expected Behavior

BetterDiscord should provide a tool that allows automated repair of BetterDiscord after a Discord update (that should run without the need of additional user interaction). Ideally what plugins are active should be remembered as well.

Discord is updated weekly, I dont think its reasonable to assume most users to be willing to re-run an install wizard and re-active plugins manually this often.

### Steps To Reproduce

-

### Operating System

Windows

### Discord Version

Stable

### BetterDiscord Version

1.9.5 and below

**Comment by ItMeSarah** [comment]
That is completely normal. Injection gets removed when Discord updates because it replaces the files and folders that BD injects into. Stable has to reinstall maybe once a month, not weekly.

**Comment by rauenzi** [comment]
See above and it's also a duplicate of #1231. You should not have to re-enable plugins or themes that data is preserved through reinstall, plugins get disabled if you use repair instead of install. Generally host updates on Discord stable only happen a few times a year.

**Comment by ItsCubeTime** [comment]
> host updates on Discord stable only happen a few times a year.

On my device (win11 v10.0.22000.2538) with a version of discord installed directly from [discord.com](https://discord.com/) I get updates that uninstall betterdiscord more than weekly.

Used to be the same on my previous PC which ran win10

**Comment by Riddim-GLiTCH** [comment]
Then your Discord app decides that it should refresh its data. There is not much you can do other than actually reinstalling BetterDiscord after it has done so. There is nothing you can do to automate this task. as they cannot detect when Discord does do a refresh. And even if they could, Betterdiscord works by modifying the core to load BetterDiscords own ASAR, Which is a change that can only be applied after restarting the app as the core is one of the first things that are loaded. And BD loads only AFTER the discord app has fully loaded.

If it really bothers you to have to reinstall BetterDiscord Every single time this happens, I can only reccomend you take a look at [Knewests BetterDiscord Install Scripts](https://gist.github.com/Knewest/ee59d3960e18e6d813c9221b54b36ab1). They should provide you with a single click reinstall of BetterDiscord.

**Comment by ItsCubeTime** [comment]
Thanks for the link @Riddim-GLiTCH . I will try it next time my injection fails.
